cmd/locations.go

```diff
@@ -0,0 +1,82 @@
+package cmd
+
+import (
+	"context"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/docker/docker/builder/remotecontext/urlutil"
+	"github.com/pkg/errors"
+	"github.com/rs/zerolog/log"
+
+	"github.com/zapier/kubechecks/pkg"
+	"github.com/zapier/kubechecks/pkg/container"
+	"github.com/zapier/kubechecks/pkg/git"
+)
+
+func processLocations(ctx context.Context, ctr container.Container, locations []string) error {
+	for index, location := range locations {
+		if newLocation, err := maybeCloneGitUrl(ctx, ctr.RepoManager, location, ctr.VcsClient.Username()); err != nil {
+			return errors.Wrapf(err, "failed to clone %q", location)
+		} else if newLocation != "" {
+			locations[index] = newLocation
+		}
+	}
+
+	return nil
+}
+
+type cloner interface {
+	Clone(ctx context.Context, cloneUrl, branchName string) (*git.Repo, error)
+}
+
+var ErrCannotUseQueryWithFilePath = errors.New("relative and absolute file paths cannot have query parameters")
+
+func maybeCloneGitUrl(ctx context.Context, repoManager cloner, location, vcsUsername string) (string, error) {
+	result := strings.SplitN(location, "?", 2)
+	if !urlutil.IsGitURL(result[0]) {
+		if len(result) > 1 {
+			return "", ErrCannotUseQueryWithFilePath
+		}
+		return result[0], nil
+	}
+
+	repoUrl, query, err := pkg.NormalizeRepoUrl(location)
+	if err != nil {
+		return "", errors.Wrapf(err, "invalid git url: %q", location)
+	}
+	cloneUrl := repoUrl.CloneURL(vcsUsername)
+
+	repo, err := repoManager.Clone(ctx, cloneUrl, query.Get("branch"))
+	if err != nil {
+		return "", errors.Wrap(err, "failed to clone")
+	}
+
+	go func() {
+		tick := time.Tick(time.Minute * 5)
```

Comment:
My concern was that, yes, commits don't show up that often, but when you do make changes, it's probably because you want to immediately take advantage of them. Up to 5 minutes is a reasonable amount of time to wait, more than that and you'll have to go cycle the pods in order to see the changes. Given that it's a pretty cheap operation (connect to git, check the hash, nothing's changed, disconnect), I thought 5 minutes was a pretty reasonable amount of time. 

... that said, it's easy to make it configurable. I'll do that. Thanks!

Reply:
... after merging this though, this PR is getting large.